src/reducer.test.js

```diff
@@ -1,23 +1,28 @@
+import given from 'given2';
+
 import reducer from './reducer';
 
 import {
   setRegions,
   setCategories,
   setRestaurants,
+  setRestaurantDetail,
   selectRegion,
   selectCategory,
 } from './actions';
 
 describe('reducer', () => {
-  context('when previous state is undefined', () => {
-    const initialState = {
-      regions: [],
-      categories: [],
-      restaurants: [],
-      selectedRegion: null,
-      selectedCategory: null,
-    };
+  const initialState = {
+    regions: [],
+    categories: [],
+    restaurants: [],
+    restaurantDetail: null,
+    selectedRegion: null,
+    selectedCategory: null,
+    selectedRestaurant: null,
+  };
```

Comment:
```suggestion
  const initialState = {
    regions: [],
    categories: [],
    restaurants: [],
    restaurantDetail: given.restaurantDetail,
    selectedRegion: given.selectedRegion,
    ...
  };
```

같이 변화되는 부분만 given 으로 주셔도 될거에요.

Reply:
주말에도 리뷰해주셔서 감사합니다!
```
    restaurantDetail: given.restaurantDetail,
    selectedRegion: given.selectedRegion,
```
을 써서 값을 컨트롤 해주려면 initailState도 
```
  given('initialState', () => ({
```
로 써줘야되는 것 같은데요, 혹시 suggestion처럼 바로 
```
    restaurantDetail: given.restaurantDetail,
    selectedRegion: given.selectedRegion,
```
로 써도 값을 넣어주는 방법이 있을까요?

Reply:
필요하다면 처음에 `given('rerstaurantDetail', () => '값');` 처럼 지정해 주시고 context 에 따라 변경해 주는 방법도 있을거에요.
테스트별로 `rerstaurantDetail` 등이 항상 값이 필요한게 아닐거라서 필요할 때 넣어주시면 될 것 같아요.